internal/buffer/buffer.go

```diff
@@ -734,16 +748,53 @@ func (b *Buffer) UpdateRules() {
 		}
 
 		if ft == "unknown" || ft == "" {
-			if highlight.MatchFiletype(header.FtDetect, b.Path, b.lines[0].data) {
-				syntaxFile = f.Name()
-				break
+			if highlight.MatchFileName(header, b.Path) {
+				syntaxFiles = append(syntaxFiles, SyntaxFileBuffer{header, f.Name(), nil})
 			}
 		} else if header.FileType == ft {
 			syntaxFile = f.Name()
 			break
 		}
 	}
 
+	if syntaxFile == "" {
+		length := len(syntaxFiles)
+		if length > 0 {
+			signatureMatch := false
+			if length > 1 {
+				detectlimit := util.IntOpt(b.Settings["detectlimit"])
+				lineCount := len(b.lines)
+				limit := 0
+				if detectlimit > 0 {
+					if lineCount < detectlimit {
+						limit = lineCount
+					} else {
+						limit = detectlimit
+					}
+				} else {
+					limit = lineCount
+				}
```

Comment:
I see you've moved the detectlimit stuff from the next commit to this one (I guess not on purpose). Could you move it back? :)

Reply:
Argh, failed at merge conflict resolution...seems it's getting late.